npc/re/merchants/prt_market.txt

```diff
@@ -0,0 +1,245 @@
+		setarray .@warp_x, 161, 285, 38, 161;
+		setarray .@warp_y, 143, 162, 162, 287;
+	}
+	else if (.@npc_name$ == "Guide#mk_prt4") {
+		.@menu$[0] = "Move to the Center Market Street";
+		.@menu$[1] = "Move to the South Market Street";
+		.@menu$[2] = "Move to the West Market Street";
+		.@menu$[3] = "Move to the East Market Street";
+		setarray .@direction$[0], "to the Center", "to the South", "to the West", "to the East";
+		setarray .@warp_x, 161, 161, 38, 285;
+		setarray .@warp_y, 143, 39, 162, 162;
+	} else {
+		.@menu$[0] = "Move to the East Market Street";
+		.@menu$[1] = "Move to the South Market Street";
+		.@menu$[2] = "Move to the West Market Street";
+		.@menu$[3] = "Move to the North Market Street";
+		setarray .@direction$[0], "to the East", "to the South", "to the West", "to the North";
+		setarray .@warp_x, 285, 161, 38, 161;
+		setarray .@warp_y, 162, 39, 162, 287;
+	}
+
+	.@menu = select(.@menu$[0], .@menu$[1], .@menu$[2], .@menu$[3], "~ Move to the Center of Prontera", "Cancel");
+	if (.@menu == 5) {
+		mes "[Guide]";
+		if (.@npc_name$ == "Guide#mk_prt1") {
+			set .@direction$, "to the East of";
+			set .@warp_x, 275;
+			set .@warp_y, 204;
+		}
+		else if (.@npc_name$ == "Guide#mk_prt2") {
+			set .@direction$, "to the West of";
+			set .@warp_x, 36;
+			set .@warp_y, 204;
+		}
+		else if (.@npc_name$ == "Guide#mk_prt3") {
+			set .@direction$, "to the South of";
+			set .@warp_x, 155;
+			set .@warp_y, 35;
+		}
+		else if (.@npc_name$ == "Guide#mk_prt4") {
+			set .@direction$, "to the North of";
+			set .@warp_x, 155;
+			set .@warp_y, 323;
+		} else {
+			set .@direction$, "to the Central Fountain";
+			set .@warp_x, 155;
+			set .@warp_y, 183;
+		}
+		mes "You will be moved " + .@direction$ + " the Center of Prontera.";
+		close2;
+		warp "prontera",.@warp_x,.@warp_y;
+		end;
+	} else if (.@menu == 6) {
+		mes "[Guide]";
+		mes "Please come back if you need to move within the market street or to the center of Prontera.";
+		close3;
+	} else {
+		mes "[Guide]";
+		mes "I will guide you " + .@direction$[.@menu - 1] + ".";
+		close2;
+		warp "prt_mk", .@warp_x[.@menu - 1], .@warp_y[.@menu - 1];
+		end;
+	}
+}
+prt_mk,255,304,3	script	Unmanned Storage Box 1#mk_prt1-1	4_STEELBOX,{
+	mes "This is an Unmanned Storage Box.";
+	mes "It is connected to Kafra Service's Storage 1.";
+	mes "Please insert 50z to use the Storage.";
+	if (!callfunc("F_CanOpenStorage")) {
+		mes "^4d4dffTo use the Storage Service, you need to raise the Novice's Basic Skill to 6.^000000";
+		close;
+	}
+	if (Zeny < 50) {
+		mes "The Storage access fee is 50z.";
+		close;
+	}
+	next;
+	switch( select( "Insert 50z", "Cancel" )) {
+		case 1:
+			Zeny = Zeny - 50;
+			if(getstatus(SC_PREMIUM_STORAGEBOOST))
+				openstorage2 4,STOR_MODE_GET|STOR_MODE_PUT;
+			else
+				openstorage;
+			close;
+		case 2:
+			mes "Thank you for using our service.";
+			close;
+	}
+}
+prt_mk,258,304,3	script	Unmanned Storage Box 2#mk_prt1-2	4_STEELBOX,{
```

Comment:
Can we use some callfunc magic for these NPCs to remove the redundant text? The only differences seem to really be the storage name and the openstorage types.